src/gil.rs

```diff
@@ -577,54 +588,51 @@ mod tests {
 
     #[test]
     fn test_pyobject_drop_with_gil_decreases_refcnt() {
-        #[allow(deprecated)]
-        let gil = Python::acquire_gil();
-        let py = gil.python();
-        let obj = get_object(py);
-        // Ensure that obj does not get freed
-        let _ref = obj.clone_ref(py);
-        let obj_ptr = obj.as_ptr();
+        Python::with_gil(|py| {
+            let obj = get_object(py);
 
-        unsafe {
-            {
-                assert_eq!(owned_object_count(), 0);
-                assert_eq!(ffi::Py_REFCNT(obj_ptr), 2);
-            }
+            // Create a reference to drop with the GIL.
+            let reference = obj.clone_ref(py);
 
-            // With the GIL held, obj can be dropped immediately
-            drop(obj);
-            assert_eq!(ffi::Py_REFCNT(obj_ptr), 1);
-        }
+            assert_eq!(obj.get_refcnt(py), 2);
+            assert!(pool_not_dirty());
+
+            // With the GIL held, reference cound will be decreased immediately.
+            drop(reference);
+
+            assert_eq!(obj.get_refcnt(py), 1);
+            assert!(pool_not_dirty());
+        });
     }
 
     #[test]
     fn test_pyobject_drop_without_gil_doesnt_decrease_refcnt() {
-        #[allow(deprecated)]
-        let gil = Python::acquire_gil();
-        let py = gil.python();
-        let obj = get_object(py);
-        // Ensure that obj does not get freed
-        let _ref = obj.clone_ref(py);
-        let obj_ptr = obj.as_ptr();
-
-        unsafe {
-            {
-                assert_eq!(owned_object_count(), 0);
-                assert_eq!(ffi::Py_REFCNT(obj_ptr), 2);
-            }
-
-            // Without the GIL held, obj cannot be dropped until the next GIL acquire
-            drop(gil);
-            drop(obj);
-            assert_eq!(ffi::Py_REFCNT(obj_ptr), 2);
+        let obj = Python::with_gil(|py| {
+            let obj = get_object(py);
+            // Create a reference to drop without the GIL.
+            let reference = obj.clone_ref(py);
+
+            assert_eq!(obj.get_refcnt(py), 2);
+            assert!(pool_not_dirty());
+
+            // Drop reference in a separate thread which doesn't have the GIL.
+            std::thread::spawn(move || drop(reference)).join().unwrap();
```

Comment:
I suspect this test function needs a

```rust
#[cfg(not(target_arch = "wasm32"))] // We are building wasm Python with pthreads disabled
```

attribute like the other tests which spawn threads.